Submitted By: Joe Ciccone <jciccone@linuxfromscratch.org>
Date: 2006-07-22
Upstream Status: Not Submitted
Origin: Joe Ciccone
Description: Adds a Makefile to assit with the cross-compilation of perl.

             Archs supported: arm x86

             Usage:
               cd Cross
               make ARCH=[arch] CROSS_COMPILE=[cross-tools-prefix]
               make DESTDIR=[optional:fakeroot] install

diff -Naur perl-5.8.8.orig/Cross/config perl-5.8.8/Cross/config
--- perl-5.8.8.orig/Cross/config	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/config	1969-12-31 19:00:00.000000000 -0500
@@ -1,23 +0,0 @@
-##############################################################################
-#
-# This is the configuration file used to specify the target platform and
-# required level of debug or optimisation.
-#
-##############################################################################
-
-### Target Arch
-ARCH = arm
-#ARCH = i386-pc
-#ARCH = sh4
-#ARCH = mips
-#ARCH = mipsel
-#ARCH = ppc
-
-## Specific arm
-CONFIG_TARGET_ARM_SA11X0 = 1
-#CONFIG_TARGET_ARM_XSCALE = 1
-
-### Target OS
-OS = linux
-#OS = solaris2.8
-
diff -Naur perl-5.8.8.orig/Cross/config.sh-arm-linux perl-5.8.8/Cross/config.sh-arm-linux
--- perl-5.8.8.orig/Cross/config.sh-arm-linux	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/config.sh-arm-linux	2006-07-22 18:17:35.000000000 -0400
@@ -8,9 +8,9 @@
 
 # Package name      : perl5
 # Source directory  : .
-# Configuration time: Wed Sep  3 22:24:58 EDT 2003
-# Configured by     : red
-# Target system     : linux openzaurus.criticalintegration.com 2.4.6-rmk1-np2-embedix #15 Wed Aug 6 07:49:44 UTC 2003 armv4l unknown 
+# Configuration time: Tue Jun 13 01:46:16 UTC 2006
+# Configured by     : root
+# Target system     : linux spitz 2.6.16 #1 preempt mon may 1 06:14:52 cest 2006 armv5tel gnulinux 
 
 Author=''
 Date='$Date'
@@ -36,10 +36,10 @@
 api_version='8'
 api_versionstring='5.8.0'
 ar='ar'
-archlib='/usr/lib/perl5/5.8.8/armv4l-linux'
-archlibexp='/usr/lib/perl5/5.8.8/armv4l-linux'
+archlib='/usr/lib/perl5/5.8.8/armv5tel-linux'
+archlibexp='/usr/lib/perl5/5.8.8/armv5tel-linux'
 archname64=''
-archname='armv4l-linux'
+archname='armv5tel-linux'
 archobjs=''
 asctime_r_proto='0'
 awk='awk'
@@ -55,15 +55,15 @@
 cat='cat'
 cc='cc'
 cccdlflags='-fpic'
-ccdlflags='-rdynamic -Wl,-rpath,/usr/lib/perl5/5.8.8/armv4l-linux/CORE'
-ccflags='-fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
+ccdlflags='-Wl,-E'
+ccflags='-fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
 ccflags_uselargefiles='-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
 ccname='gcc'
-ccsymbols='__APCS_32__=1 __ARM_ARCH_3__=1 __CHAR_UNSIGNED__=1 __GNUC_MINOR__=95 __arm__=1 __linux=1 __linux__=1 __unix=1 __unix__=1 cpu=arm machine=arm system=posix system=unix'
+ccsymbols=''
 ccversion=''
-cf_by='red'
-cf_email='red@criticalintegration.com'
-cf_time='Wed Sep  3 22:24:58 EDT 2003'
+cf_by='root'
+f_email='root@spitz.ciccone'
+cf_time='Tue Jun 13 01:46:16 UTC 2006'
 charsize='1'
 chgrp=''
 chmod='chmod'
@@ -76,13 +76,13 @@
 cpio=''
 cpp='cpp'
 cpp_stuff='42'
-cppccsymbols='__ELF__=1 __GNUC__=2 linux=1 unix=1'
-cppflags='-fno-strict-aliasing -I/usr/local/include'
+cppccsymbols=''
+cppflags='-fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include'
 cpplast='-'
 cppminus='-'
 cpprun='cc -E'
 cppstdin='cc -E'
-cppsymbols='_FILE_OFFSET_BITS=64 __GLIBC__=2 __GLIBC_MINOR__=2 __GNUC_MINOR__=95 __GNU_LIBRARY__=6 _LARGEFILE_SOURCE=1 _POSIX_C_SOURCE=199506 _POSIX_SOURCE=1 __STDC__=1 __USE_BSD=1 __USE_FILE_OFFSET64=1 __USE_LARGEFILE=1 __USE_MISC=1 __USE_POSIX=1 __USE_POSIX199309=1 __USE_POSIX199506=1 __USE_POSIX2=1 __USE_SVID=1 __linux=1 __linux__=1 __unix=1 __unix__=1'
+cppsymbols='__ELF__=1 _FILE_OFFSET_BITS=64 __GLIBC__=2 __GLIBC_MINOR__=4 __GNUC__=4 __GNUC_MINOR__=1 __GNU_LIBRARY__=6 _LARGEFILE_SOURCE=1 _POSIX_C_SOURCE=200112L _POSIX_SOURCE=1 __STDC__=1 __USE_BSD=1 __USE_FILE_OFFSET64=1 __USE_LARGEFILE=1 __USE_MISC=1 __USE_POSIX=1 __USE_POSIX199309=1 __USE_POSIX199506=1 __USE_POSIX2=1 __USE_SVID=1 linux=1 __linux=1 __linux__=1 unix=1 __unix=1 __unix__=1'
 crypt_r_proto='0'
 cryptlib=''
 csh='csh'
@@ -111,13 +111,13 @@
 d_asctime_r='undef'
 d_atolf='undef'
 d_atoll='define'
-d_attribute_format='undef'
-d_attribute_malloc='undef'
-d_attribute_nonnull='undef'
-d_attribute_noreturn='undef'
-d_attribute_pure='undef'
-d_attribute_unused='undef'
-d_attribute_warn_unused_result='undef'
+d_attribute_format='define'
+d_attribute_malloc='define'
+d_attribute_nonnull='define'
+d_attribute_noreturn='define'
+d_attribute_pure='define'
+d_attribute_unused='define'
+d_attribute_warn_unused_result='define'
 d_bcmp='define'
 d_bcopy='define'
 d_bsd='undef'
@@ -126,7 +126,7 @@
 d_bzero='define'
 d_casti32='define'
 d_castneg='define'
-d_charvspr='undef'
+d_charvspr='define'
 d_chown='define'
 d_chroot='define'
 d_chsize='undef'
@@ -138,7 +138,7 @@
 d_copysignl='define'
 d_crypt='define'
 d_crypt_r='undef'
-d_csh='define'
+d_csh='undef'
 d_ctermid_r='undef'
 d_ctime_r='undef'
 d_cuserid='define'
@@ -154,7 +154,7 @@
 d_drand48_r='undef'
 d_drand48proto='define'
 d_dup2='define'
-d_eaccess='undef'
+d_eaccess='define'
 d_endgrent='define'
 d_endgrent_r='undef'
 d_endhent='define'
@@ -169,7 +169,7 @@
 d_endservent_r='undef'
 d_eofnblk='define'
 d_eunice='undef'
-d_faststdio='define'
+d_faststdio='undef'
 d_fchdir='define'
 d_fchmod='define'
 d_fchown='define'
@@ -274,6 +274,7 @@
 d_killpg='define'
 d_lchown='define'
 d_ldbl_dig='define'
+d_libm_lib_version='define'
 d_link='define'
 d_localtime_r='undef'
 d_locconv='define'
@@ -321,7 +322,7 @@
 d_nice='define'
 d_nl_langinfo='define'
 d_nv_preserves_uv='define'
-d_nv_zero_is_allbits_zero='undef'
+d_nv_zero_is_allbits_zero='define'
 d_off64_t='undef'
 d_old_pthread_create_joinable='undef'
 d_oldpthreads='undef'
@@ -334,7 +335,7 @@
 d_pipe='define'
 d_poll='define'
 d_portable='define'
-d_procselfexe='define'
+d_procselfexe='undef'
 d_pthread_atfork='undef'
 d_pthread_attr_setscope='define'
 d_pthread_yield='undef'
@@ -416,9 +417,8 @@
 d_sigaction='define'
 d_sigprocmask='define'
 d_sigsetjmp='define'
-d_snprintf='undef'
-d_sockatmark='undef'
-d_sockatmarkproto='undef'
+d_sockatmark='define'
+d_sockatmarkproto='define'
 d_socket='define'
 d_socklen_t='define'
 d_sockpair='define'
@@ -434,12 +434,12 @@
 d_statfs_s='define'
 d_statvfs='define'
 d_stdio_cnt_lval='undef'
-d_stdio_ptr_lval='define'
+d_stdio_ptr_lval='undef'
 d_stdio_ptr_lval_nochange_cnt='undef'
-d_stdio_ptr_lval_sets_cnt='define'
+d_stdio_ptr_lval_sets_cnt='undef'
 d_stdio_stream_array='undef'
-d_stdiobase='define'
-d_stdstdio='define'
+d_stdiobase='undef'
+d_stdstdio='undef'
 d_strchr='define'
 d_strcoll='define'
 d_strctcpy='define'
@@ -484,7 +484,7 @@
 d_uname='define'
 d_union_semun='undef'
 d_unordered='undef'
-d_unsetenv='unsetenv'
+d_unsetenv='define'
 d_usleep='define'
 d_usleepproto='define'
 d_ustat='define'
@@ -498,7 +498,6 @@
 d_voidtty=''
 d_volatile='define'
 d_vprintf='define'
-d_vsnprintf='define'
 d_wait4='define'
 d_waitpid='define'
 d_wcstombs='define'
@@ -545,11 +544,11 @@
 freetype='void'
 from=':'
 full_ar='/usr/bin/ar'
-full_csh='/bin/csh'
+full_csh='csh'
 full_sed='/bin/sed'
 gccansipedantic=''
 gccosandvers=''
-gccversion='2.95.3 20010125 (prerelease)'
+gccversion='4.1.1'
 getgrent_r_proto='0'
 getgrgid_r_proto='0'
 getgrnam_r_proto='0'
@@ -577,7 +576,7 @@
 glibpth='/usr/shlib  /lib /usr/lib /usr/lib/386 /lib/386 /usr/ccs/lib /usr/ucblib /usr/local/lib '
 gmake='gmake'
 gmtime_r_proto='0'
-gnulibc_version='2.2.2'
+gnulibc_version='2.4'
 grep='grep'
 groupcat='cat /etc/group'
 groupstype='gid_t'
@@ -685,25 +684,25 @@
 inc_version_list_init='0'
 incpath=''
 inews=''
-installarchlib='./install_me_here/usr/lib/perl5/5.8.8/armv4l-linux'
-installbin='./install_me_here/usr/bin'
+installarchlib='/usr/lib/perl5/5.8.8/armv5tel-linux'
+installbin='/usr/bin'
 installhtml1dir=''
 installhtml3dir=''
-installman1dir='./install_me_here/usr/share/man/man1'
-installman3dir='./install_me_here/usr/share/man/man3'
-installprefix='./install_me_here/usr'
-installprefixexp='./install_me_here/usr'
-installprivlib='./install_me_here/usr/lib/perl5/5.8.8'
-installscript='./install_me_here/usr/bin'
-installsitearch='./install_me_here/usr/lib/perl5/site_perl/5.8.8/armv4l-linux'
-installsitebin='./install_me_here/usr/bin'
+installman1dir='/usr/share/man/man1'
+installman3dir='/usr/share/man/man3'
+installprefix='/usr'
+installprefixexp='/usr'
+installprivlib='/usr/lib/perl5/5.8.8'
+installscript='/usr/bin'
+installsitearch='/usr/lib/perl5/site_perl/5.8.8/armv5tel-linux'
+installsitebin='/usr/bin'
 installsitehtml1dir=''
 installsitehtml3dir=''
-installsitelib='./install_me_here/usr/lib/perl5/site_perl/5.8.8'
-installsiteman1dir='./install_me_here/usr/share/man/man1'
-installsiteman3dir='./install_me_here/usr/share/man/man3'
-installsitescript='./install_me_here/usr/bin'
-installstyle='./install_me_herelib/perl5'
+installsitelib='/usr/lib/perl5/site_perl/5.8.8'
+installsiteman1dir='/usr/share/man/man1'
+installsiteman3dir='/usr/share/man/man3'
+installsitescript='/usr/bin'
+installstyle='lib/perl5'
 installusrbinperl='undef'
 installvendorarch=''
 installvendorbin=''
@@ -714,7 +713,7 @@
 installvendorman3dir=''
 installvendorscript=''
 intsize='4'
-issymlink='/usr/bin/test -h'
+issymlink='/bin/test -h'
 ivdformat='"ld"'
 ivsize='4'
 ivtype='long'
@@ -727,15 +726,15 @@
 ldlibpthname='LD_LIBRARY_PATH'
 less='less'
 lib_ext='.a'
-libc='/lib/libc-2.2.2.so'
-libperl='libperl.so'
+libc='/lib/libc-2.4.so'
+libperl='libperl.a'
 libpth='/usr/local/lib /lib /usr/lib'
 libs='-lnsl -ldl -lm -lcrypt -lutil -lc'
 libsdirs=' /usr/lib'
 libsfiles=' libnsl.so libdl.so libm.so libcrypt.so libutil.so libc.so'
 libsfound=' /usr/lib/libnsl.so /usr/lib/libdl.so /usr/lib/libm.so /usr/lib/libcrypt.so /usr/lib/libutil.so /usr/lib/libc.so'
 libspath=' /usr/local/lib /lib /usr/lib'
-libswanted='sfio socket bind inet nsl nm ndbm gdbm dbm db malloc dl dld ld sun m crypt sec util c cposix posix ucb BSD'
+libswanted='sfio socket inet nsl nm ndbm gdbm dbm db malloc dl dld ld sun m crypt sec util c cposix posix ucb BSD'
 libswanted_uselargefiles=''
 line=''
 lint=''
@@ -774,10 +773,10 @@
 more='more'
 multiarch='undef'
 mv=''
-myarchname='armv4l-linux'
-mydomain='.criticalintegration.com'
-myhostname='openzaurus'
-myuname='linux openzaurus.criticalintegration.com 2.4.6-rmk1-np2-embedix #15 Wed Aug 6 07:49:44 UTC 2003 armv4l unknown '
+myarchname='armv5tel-linux'
+mydomain='.ciccone'
+myhostname='spitz'
+myuname='linux spitz 2.6.16 #1 preempt mon may 1 06:14:52 cest 2006 armv5tel gnulinux '
 n='-n'
 need_va_copy='undef'
 netdb_hlen_type='size_t'
@@ -804,17 +803,17 @@
 optimize='-O2'
 orderlib='false'
 osname='linux'
-osvers='2.4.6-rmk1-np2-embedix'
+osvers='2.6.16'
 otherlibdirs=' '
 package='perl5'
-pager='/usr/bin/less'
+pager='/bin/less -isR'
 passcat='cat /etc/passwd'
 patchlevel='8'
 path_sep=':'
-perl5='/usr/bin/perl'
+perl5=''
 perl=''
 perl_patchlevel=''
-perladmin='red@criticalintegration.com'
+perladmin='root@spitz.ciccone'
 perllibs='-lnsl -ldl -lm -lcrypt -lutil -lc'
 perlpath='/usr/bin/perl'
 pg='pg'
@@ -827,7 +826,7 @@
 prefixexp='/usr'
 privlib='/usr/lib/perl5/5.8.8'
 privlibexp='/usr/lib/perl5/5.8.8'
-procselfexe='"/proc/self/exe"'
+procselfexe=''
 prototype='define'
 ptrsize='4'
 quadkind='3'
@@ -880,15 +879,15 @@
 shortsize='2'
 shrpenv=''
 shsharp='true'
-sig_count='64'
-sig_name='ZERO HUP INT QUIT ILL TRAP ABRT BUS FPE KILL USR1 SEGV USR2 PIPE ALRM TERM STKFLT CHLD CONT STOP TSTP TTIN TTOU URG XCPU XFSZ VTALRM PROF WINCH IO PWR SYS RTMIN NUM33 NUM34 NUM35 NUM36 NUM37 NUM38 NUM39 NUM40 NUM41 NUM42 NUM43 NUM44 NUM45 NUM46 NUM47 NUM48 NUM49 NUM50 NUM51 NUM52 NUM53 NUM54 NUM55 NUM56 NUM57 NUM58 NUM59 NUM60 NUM61 NUM62 RTMAX IOT CLD POLL UNUSED '
-sig_name_init='"ZERO", "HUP", "INT", "QUIT", "ILL", "TRAP", "ABRT", "BUS", "FPE", "KILL", "USR1", "SEGV", "USR2", "PIPE", "ALRM", "TERM", "STKFLT", "CHLD", "CONT", "STOP", "TSTP", "TTIN", "TTOU", "URG", "XCPU", "XFSZ", "VTALRM", "PROF", "WINCH", "IO", "PWR", "SYS", "RTMIN", "NUM33", "NUM34", "NUM35", "NUM36", "NUM37", "NUM38", "NUM39", "NUM40", "NUM41", "NUM42", "NUM43", "NUM44", "NUM45", "NUM46", "NUM47", "NUM48", "NUM49", "NUM50", "NUM51", "NUM52", "NUM53", "NUM54", "NUM55", "NUM56", "NUM57", "NUM58", "NUM59", "NUM60", "NUM61", "NUM62", "RTMAX", "IOT", "CLD", "POLL", "UNUSED", 0'
-sig_num='0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 6 17 29 31 '
-sig_num_init='0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 6, 17, 29, 31, 0'
-sig_size='68'
+sig_count='65'
+sig_name='ZERO HUP INT QUIT ILL TRAP ABRT BUS FPE KILL USR1 SEGV USR2 PIPE ALRM TERM STKFLT CHLD CONT STOP TSTP TTIN TTOU URG XCPU XFSZ VTALRM PROF WINCH IO PWR SYS NUM32 NUM33 RTMIN NUM35 NUM36 NUM37 NUM38 NUM39 NUM40 NUM41 NUM42 NUM43 NUM44 NUM45 NUM46 NUM47 NUM48 NUM49 NUM50 NUM51 NUM52 NUM53 NUM54 NUM55 NUM56 NUM57 NUM58 NUM59 NUM60 NUM61 NUM62 NUM63 RTMAX IOT CLD POLL UNUSED '
+sig_name_init='"ZERO", "HUP", "INT", "QUIT", "ILL", "TRAP", "ABRT", "BUS", "FPE", "KILL", "USR1", "SEGV", "USR2", "PIPE", "ALRM", "TERM", "STKFLT", "CHLD", "CONT", "STOP", "TSTP", "TTIN", "TTOU", "URG", "XCPU", "XFSZ", "VTALRM", "PROF", "WINCH", "IO", "PWR", "SYS", "NUM32", "NUM33", "RTMIN", "NUM35", "NUM36", "NUM37", "NUM38", "NUM39", "NUM40", "NUM41", "NUM42", "NUM43", "NUM44", "NUM45", "NUM46", "NUM47", "NUM48", "NUM49", "NUM50", "NUM51", "NUM52", "NUM53", "NUM54", "NUM55", "NUM56", "NUM57", "NUM58", "NUM59", "NUM60", "NUM61", "NUM62", "NUM63", "RTMAX", "IOT", "CLD", "POLL", "UNUSED", 0'
+sig_num='0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 6 17 29 31 '
+sig_num_init='0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 6, 17, 29, 31, 0'
+sig_size='69'
 signal_t='void'
-sitearch='/usr/lib/perl5/site_perl/5.8.8/armv4l-linux'
-sitearchexp='/usr/lib/perl5/site_perl/5.8.8/armv4l-linux'
+sitearch='/usr/lib/perl5/site_perl/5.8.8/armv5tel-linux'
+sitearchexp='/usr/lib/perl5/site_perl/5.8.8/armv5tel-linux'
 sitebin='/usr/bin'
 sitebinexp='/usr/bin'
 sitehtml1dir=''
@@ -984,10 +983,9 @@
 useperlio='define'
 useposix='true'
 usereentrant='undef'
-userelocatableinc='undef'
 usesfio='false'
+useshrplib='false'
 usesitecustomize='undef'
-useshrplib='true'
 usesocks='undef'
 usethreads='undef'
 usevendorprefix='undef'
@@ -1031,8 +1029,14 @@
 zip='zip'
 # Configure command line arguments.
 config_arg0='Configure'
-config_args=''
-config_argc=0
+config_args='-ds -e -Dprefix=/usr -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dpager=/bin/less -isR'
+config_argc=6
+config_arg1='-ds'
+config_arg2='-e'
+config_arg3='-Dprefix=/usr'
+config_arg4='-Dman1dir=/usr/share/man/man1'
+config_arg5='-Dman3dir=/usr/share/man/man3'
+config_arg6='-Dpager=/bin/less -isR'
 PERL_REVISION=5
 PERL_VERSION=8
 PERL_SUBVERSION=8
diff -Naur perl-5.8.8.orig/Cross/config.sh-x86-linux perl-5.8.8/Cross/config.sh-x86-linux
--- perl-5.8.8.orig/Cross/config.sh-x86-linux	1969-12-31 19:00:00.000000000 -0500
+++ perl-5.8.8/Cross/config.sh-x86-linux	2006-07-22 18:18:01.000000000 -0400
@@ -0,0 +1,1047 @@
+#!/bin/sh
+#
+# This file was produced by running the Configure script. It holds all the
+# definitions figured out by Configure. Should you modify one of these values,
+# do not forget to propagate your changes by running "Configure -der". You may
+# instead choose to run each of the .SH files by yourself, or "Configure -S".
+#
+
+# Package name      : perl5
+# Source directory  : .
+# Configuration time: Sun Jul  9 12:50:37 EDT 2006
+# Configured by     : root
+# Target system     : linux lfslivecd 2.6.16.14 #1 smp sat may 13 11:48:14 gmt 2006 i686 pentium3 i386 gnulinux 
+
+Author=''
+Date='$Date'
+Header=''
+Id='$Id'
+Locker=''
+Log='$Log'
+Mcc='Mcc'
+RCSfile='$RCSfile'
+Revision='$Revision'
+Source=''
+State=''
+_a='.a'
+_exe=''
+_o='.o'
+afs='false'
+afsroot='/afs'
+alignbytes='4'
+ansi2knr=''
+aphostname='/bin/hostname'
+api_revision='5'
+api_subversion='0'
+api_version='8'
+api_versionstring='5.8.0'
+ar='ar'
+archlib='/usr/lib/perl5/5.8.8/i686-linux'
+archlibexp='/usr/lib/perl5/5.8.8/i686-linux'
+archname64=''
+archname='i686-linux'
+archobjs=''
+asctime_r_proto='0'
+awk='awk'
+baserev='5.0'
+bash=''
+bin='/usr/bin'
+binexp='/usr/bin'
+bison='bison'
+byacc='byacc'
+byteorder='1234'
+c=''
+castflags='0'
+cat='cat'
+cc='cc'
+cccdlflags='-fpic'
+ccdlflags='-Wl,-E'
+ccflags='-fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
+ccflags_uselargefiles='-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
+ccname='gcc'
+ccsymbols=''
+ccversion=''
+cf_by='root'
+cf_email='root@lfslivecd.jenacon.net'
+cf_time='Sun Jul  9 12:50:37 EDT 2006'
+charsize='1'
+chgrp=''
+chmod='chmod'
+chown=''
+clocktype='clock_t'
+comm='comm'
+compress=''
+contains='grep'
+cp='cp'
+cpio=''
+cpp='cpp'
+cpp_stuff='42'
+cppccsymbols=''
+cppflags='-fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include'
+cpplast='-'
+cppminus='-'
+cpprun='cc -E'
+cppstdin='cc -E'
+cppsymbols='__ELF__=1 _FILE_OFFSET_BITS=64 __GLIBC__=2 __GLIBC_MINOR__=4 __GNUC__=4 __GNUC_MINOR__=1 __GNU_LIBRARY__=6 _LARGEFILE_SOURCE=1 _POSIX_C_SOURCE=200112L _POSIX_SOURCE=1 __STDC__=1 __USE_BSD=1 __USE_FILE_OFFSET64=1 __USE_LARGEFILE=1 __USE_MISC=1 __USE_POSIX=1 __USE_POSIX199309=1 __USE_POSIX199506=1 __USE_POSIX2=1 __USE_SVID=1 i386=1 __i386=1 __i386__=1 linux=1 __linux=1 __linux__=1 unix=1 __unix=1 __unix__=1'
+crypt_r_proto='0'
+cryptlib=''
+csh='csh'
+ctermid_r_proto='0'
+ctime_r_proto='0'
+d_Gconvert='gcvt((x),(n),(b))'
+d_PRIEUldbl='define'
+d_PRIFUldbl='define'
+d_PRIGUldbl='define'
+d_PRIXU64='define'
+d_PRId64='define'
+d_PRIeldbl='define'
+d_PRIfldbl='define'
+d_PRIgldbl='define'
+d_PRIi64='define'
+d_PRIo64='define'
+d_PRIu64='define'
+d_PRIx64='define'
+d_SCNfldbl='define'
+d__fwalk='undef'
+d_access='define'
+d_accessx='undef'
+d_aintl='undef'
+d_alarm='define'
+d_archlib='define'
+d_asctime_r='undef'
+d_atolf='undef'
+d_atoll='define'
+d_attribute_format='define'
+d_attribute_malloc='define'
+d_attribute_nonnull='define'
+d_attribute_noreturn='define'
+d_attribute_pure='define'
+d_attribute_unused='define'
+d_attribute_warn_unused_result='define'
+d_bcmp='define'
+d_bcopy='define'
+d_bsd='undef'
+d_bsdgetpgrp='undef'
+d_bsdsetpgrp='undef'
+d_bzero='define'
+d_casti32='undef'
+d_castneg='define'
+d_charvspr='define'
+d_chown='define'
+d_chroot='define'
+d_chsize='undef'
+d_class='undef'
+d_clearenv='define'
+d_closedir='define'
+d_cmsghdr_s='define'
+d_const='define'
+d_copysignl='define'
+d_crypt='define'
+d_crypt_r='undef'
+d_csh='undef'
+d_ctermid_r='undef'
+d_ctime_r='undef'
+d_cuserid='define'
+d_dbl_dig='define'
+d_dbminitproto='undef'
+d_difftime='define'
+d_dirfd='define'
+d_dirnamlen='undef'
+d_dlerror='define'
+d_dlopen='define'
+d_dlsymun='undef'
+d_dosuid='undef'
+d_drand48_r='undef'
+d_drand48proto='define'
+d_dup2='define'
+d_eaccess='define'
+d_endgrent='define'
+d_endgrent_r='undef'
+d_endhent='define'
+d_endhostent_r='undef'
+d_endnent='define'
+d_endnetent_r='undef'
+d_endpent='define'
+d_endprotoent_r='undef'
+d_endpwent='define'
+d_endpwent_r='undef'
+d_endsent='define'
+d_endservent_r='undef'
+d_eofnblk='define'
+d_eunice='undef'
+d_faststdio='undef'
+d_fchdir='define'
+d_fchmod='define'
+d_fchown='define'
+d_fcntl='define'
+d_fcntl_can_lock='define'
+d_fd_macros='define'
+d_fd_set='define'
+d_fds_bits='undef'
+d_fgetpos='define'
+d_finite='define'
+d_finitel='define'
+d_flexfnam='define'
+d_flock='define'
+d_flockproto='define'
+d_fork='define'
+d_fp_class='undef'
+d_fpathconf='define'
+d_fpclass='undef'
+d_fpclassify='undef'
+d_fpclassl='undef'
+d_fpos64_t='undef'
+d_frexpl='define'
+d_fs_data_s='undef'
+d_fseeko='define'
+d_fsetpos='define'
+d_fstatfs='define'
+d_fstatvfs='define'
+d_fsync='define'
+d_ftello='define'
+d_ftime='undef'
+d_futimes='define'
+d_getcwd='define'
+d_getespwnam='undef'
+d_getfsstat='undef'
+d_getgrent='define'
+d_getgrent_r='undef'
+d_getgrgid_r='undef'
+d_getgrnam_r='undef'
+d_getgrps='define'
+d_gethbyaddr='define'
+d_gethbyname='define'
+d_gethent='define'
+d_gethname='define'
+d_gethostbyaddr_r='undef'
+d_gethostbyname_r='undef'
+d_gethostent_r='undef'
+d_gethostprotos='define'
+d_getitimer='define'
+d_getlogin='define'
+d_getlogin_r='undef'
+d_getmnt='undef'
+d_getmntent='define'
+d_getnbyaddr='define'
+d_getnbyname='define'
+d_getnent='define'
+d_getnetbyaddr_r='undef'
+d_getnetbyname_r='undef'
+d_getnetent_r='undef'
+d_getnetprotos='define'
+d_getpagsz='define'
+d_getpbyname='define'
+d_getpbynumber='define'
+d_getpent='define'
+d_getpgid='define'
+d_getpgrp2='undef'
+d_getpgrp='define'
+d_getppid='define'
+d_getprior='define'
+d_getprotobyname_r='undef'
+d_getprotobynumber_r='undef'
+d_getprotoent_r='undef'
+d_getprotoprotos='define'
+d_getprpwnam='undef'
+d_getpwent='define'
+d_getpwent_r='undef'
+d_getpwnam_r='undef'
+d_getpwuid_r='undef'
+d_getsbyname='define'
+d_getsbyport='define'
+d_getsent='define'
+d_getservbyname_r='undef'
+d_getservbyport_r='undef'
+d_getservent_r='undef'
+d_getservprotos='define'
+d_getspnam='define'
+d_getspnam_r='undef'
+d_gettimeod='define'
+d_gmtime_r='undef'
+d_gnulibc='define'
+d_grpasswd='define'
+d_hasmntopt='define'
+d_htonl='define'
+d_ilogbl='define'
+d_index='undef'
+d_inetaton='define'
+d_int64_t='define'
+d_isascii='define'
+d_isfinite='undef'
+d_isinf='define'
+d_isnan='define'
+d_isnanl='define'
+d_killpg='define'
+d_lchown='define'
+d_ldbl_dig='define'
+d_libm_lib_version='define'
+d_link='define'
+d_localtime_r='undef'
+d_locconv='define'
+d_lockf='define'
+d_longdbl='define'
+d_longlong='define'
+d_lseekproto='define'
+d_lstat='define'
+d_madvise='define'
+d_malloc_good_size='undef'
+d_malloc_size='undef'
+d_mblen='define'
+d_mbstowcs='define'
+d_mbtowc='define'
+d_memchr='define'
+d_memcmp='define'
+d_memcpy='define'
+d_memmove='define'
+d_memset='define'
+d_mkdir='define'
+d_mkdtemp='define'
+d_mkfifo='define'
+d_mkstemp='define'
+d_mkstemps='undef'
+d_mktime='define'
+d_mmap='define'
+d_modfl='define'
+d_modfl_pow32_bug='undef'
+d_modflproto='define'
+d_mprotect='define'
+d_msg='define'
+d_msg_ctrunc='define'
+d_msg_dontroute='define'
+d_msg_oob='define'
+d_msg_peek='define'
+d_msg_proxy='define'
+d_msgctl='define'
+d_msgget='define'
+d_msghdr_s='define'
+d_msgrcv='define'
+d_msgsnd='define'
+d_msync='define'
+d_munmap='define'
+d_mymalloc='undef'
+d_nice='define'
+d_nl_langinfo='define'
+d_nv_preserves_uv='define'
+d_nv_zero_is_allbits_zero='define'
+d_off64_t='undef'
+d_old_pthread_create_joinable='undef'
+d_oldpthreads='undef'
+d_oldsock='undef'
+d_open3='define'
+d_pathconf='define'
+d_pause='define'
+d_perl_otherlibdirs='undef'
+d_phostname='undef'
+d_pipe='define'
+d_poll='define'
+d_portable='define'
+d_procselfexe='define'
+d_pthread_atfork='undef'
+d_pthread_attr_setscope='define'
+d_pthread_yield='undef'
+d_pwage='undef'
+d_pwchange='undef'
+d_pwclass='undef'
+d_pwcomment='undef'
+d_pwexpire='undef'
+d_pwgecos='define'
+d_pwpasswd='define'
+d_pwquota='undef'
+d_qgcvt='define'
+d_quad='define'
+d_random_r='undef'
+d_readdir64_r='undef'
+d_readdir='define'
+d_readdir_r='undef'
+d_readlink='define'
+d_readv='define'
+d_recvmsg='define'
+d_rename='define'
+d_rewinddir='define'
+d_rmdir='define'
+d_safebcpy='undef'
+d_safemcpy='undef'
+d_sanemcmp='define'
+d_sbrkproto='define'
+d_scalbnl='define'
+d_sched_yield='define'
+d_scm_rights='define'
+d_seekdir='define'
+d_select='define'
+d_sem='define'
+d_semctl='define'
+d_semctl_semid_ds='define'
+d_semctl_semun='define'
+d_semget='define'
+d_semop='define'
+d_sendmsg='define'
+d_setegid='define'
+d_seteuid='define'
+d_setgrent='define'
+d_setgrent_r='undef'
+d_setgrps='define'
+d_sethent='define'
+d_sethostent_r='undef'
+d_setitimer='define'
+d_setlinebuf='define'
+d_setlocale='define'
+d_setlocale_r='undef'
+d_setnent='define'
+d_setnetent_r='undef'
+d_setpent='define'
+d_setpgid='define'
+d_setpgrp2='undef'
+d_setpgrp='define'
+d_setprior='define'
+d_setproctitle='undef'
+d_setprotoent_r='undef'
+d_setpwent='define'
+d_setpwent_r='undef'
+d_setregid='define'
+d_setresgid='define'
+d_setresuid='define'
+d_setreuid='define'
+d_setrgid='undef'
+d_setruid='undef'
+d_setsent='define'
+d_setservent_r='undef'
+d_setsid='define'
+d_setvbuf='define'
+d_sfio='undef'
+d_shm='define'
+d_shmat='define'
+d_shmatprototype='define'
+d_shmctl='define'
+d_shmdt='define'
+d_shmget='define'
+d_sigaction='define'
+d_sigprocmask='define'
+d_sigsetjmp='define'
+d_sockatmark='define'
+d_sockatmarkproto='define'
+d_socket='define'
+d_socklen_t='define'
+d_sockpair='define'
+d_socks5_init='undef'
+d_sprintf_returns_strlen='define'
+d_sqrtl='define'
+d_srand48_r='undef'
+d_srandom_r='undef'
+d_sresgproto='undef'
+d_sresuproto='undef'
+d_statblks='define'
+d_statfs_f_flags='undef'
+d_statfs_s='define'
+d_statvfs='define'
+d_stdio_cnt_lval='undef'
+d_stdio_ptr_lval='undef'
+d_stdio_ptr_lval_nochange_cnt='undef'
+d_stdio_ptr_lval_sets_cnt='undef'
+d_stdio_stream_array='undef'
+d_stdiobase='undef'
+d_stdstdio='undef'
+d_strchr='define'
+d_strcoll='define'
+d_strctcpy='define'
+d_strerrm='strerror(e)'
+d_strerror='define'
+d_strerror_r='undef'
+d_strftime='define'
+d_strlcat='undef'
+d_strlcpy='undef'
+d_strtod='define'
+d_strtol='define'
+d_strtold='define'
+d_strtoll='define'
+d_strtoq='define'
+d_strtoul='define'
+d_strtoull='define'
+d_strtouq='define'
+d_strxfrm='define'
+d_suidsafe='undef'
+d_symlink='define'
+d_syscall='define'
+d_syscallproto='define'
+d_sysconf='define'
+d_sysernlst=''
+d_syserrlst='define'
+d_system='define'
+d_tcgetpgrp='define'
+d_tcsetpgrp='define'
+d_telldir='define'
+d_telldirproto='define'
+d_time='define'
+d_times='define'
+d_tm_tm_gmtoff='define'
+d_tm_tm_zone='define'
+d_tmpnam_r='undef'
+d_truncate='define'
+d_ttyname_r='undef'
+d_tzname='define'
+d_u32align='undef'
+d_ualarm='define'
+d_umask='define'
+d_uname='define'
+d_union_semun='undef'
+d_unordered='undef'
+d_unsetenv='define'
+d_usleep='define'
+d_usleepproto='define'
+d_ustat='define'
+d_vendorarch='undef'
+d_vendorbin='undef'
+d_vendorlib='undef'
+d_vendorscript='undef'
+d_vfork='undef'
+d_void_closedir='undef'
+d_voidsig='define'
+d_voidtty=''
+d_volatile='define'
+d_vprintf='define'
+d_wait4='define'
+d_waitpid='define'
+d_wcstombs='define'
+d_wctomb='define'
+d_writev='define'
+d_xenix='undef'
+date='date'
+db_hashtype='u_int32_t'
+db_prefixtype='size_t'
+db_version_major=''
+db_version_minor=''
+db_version_patch=''
+defvoidused='15'
+direntrytype='struct dirent'
+dlext='so'
+dlsrc='dl_dlopen.xs'
+doublesize='8'
+drand01='drand48()'
+drand48_r_proto='0'
+dynamic_ext='B ByteLoader Cwd Data/Dumper Devel/DProf Devel/PPPort Devel/Peek Digest/MD5 Encode Fcntl File/Glob Filter/Util/Call I18N/Langinfo IO IPC/SysV List/Util MIME/Base64 Opcode POSIX PerlIO/encoding PerlIO/scalar PerlIO/via SDBM_File Socket Storable Sys/Hostname Sys/Syslog Time/HiRes Unicode/Normalize XS/APItest XS/Typemap attrs re threads threads/shared'
+eagain='EAGAIN'
+ebcdic='undef'
+echo='echo'
+egrep='egrep'
+emacs=''
+endgrent_r_proto='0'
+endhostent_r_proto='0'
+endnetent_r_proto='0'
+endprotoent_r_proto='0'
+endpwent_r_proto='0'
+endservent_r_proto='0'
+eunicefix=':'
+exe_ext=''
+expr='expr'
+extensions='B ByteLoader Cwd Data/Dumper Devel/DProf Devel/PPPort Devel/Peek Digest/MD5 Encode Fcntl File/Glob Filter/Util/Call I18N/Langinfo IO IPC/SysV List/Util MIME/Base64 Opcode POSIX PerlIO/encoding PerlIO/scalar PerlIO/via SDBM_File Socket Storable Sys/Hostname Sys/Syslog Time/HiRes Unicode/Normalize XS/APItest XS/Typemap attrs re threads threads/shared Errno'
+extras=''
+fflushNULL='define'
+fflushall='undef'
+find=''
+firstmakefile='makefile'
+flex=''
+fpossize='16'
+fpostype='fpos_t'
+freetype='void'
+from=':'
+full_ar='/usr/bin/ar'
+full_csh='csh'
+full_sed='/bin/sed'
+gccansipedantic=''
+gccosandvers=''
+gccversion='4.1.1'
+getgrent_r_proto='0'
+getgrgid_r_proto='0'
+getgrnam_r_proto='0'
+gethostbyaddr_r_proto='0'
+gethostbyname_r_proto='0'
+gethostent_r_proto='0'
+getlogin_r_proto='0'
+getnetbyaddr_r_proto='0'
+getnetbyname_r_proto='0'
+getnetent_r_proto='0'
+getprotobyname_r_proto='0'
+getprotobynumber_r_proto='0'
+getprotoent_r_proto='0'
+getpwent_r_proto='0'
+getpwnam_r_proto='0'
+getpwuid_r_proto='0'
+getservbyname_r_proto='0'
+getservbyport_r_proto='0'
+getservent_r_proto='0'
+getspnam_r_proto='0'
+gidformat='"lu"'
+gidsign='1'
+gidsize='4'
+gidtype='gid_t'
+glibpth='/usr/shlib  /lib /usr/lib /usr/lib/386 /lib/386 /usr/ccs/lib /usr/ucblib /usr/local/lib '
+gmake='gmake'
+gmtime_r_proto='0'
+gnulibc_version='2.4'
+grep='grep'
+groupcat='cat /etc/group'
+groupstype='gid_t'
+gzip='gzip'
+h_fcntl='false'
+h_sysfile='true'
+hint='recommended'
+hostcat='cat /etc/hosts'
+html1dir=' '
+html1direxp=''
+html3dir=' '
+html3direxp=''
+i16size='2'
+i16type='short'
+i32size='4'
+i32type='long'
+i64size='8'
+i64type='long long'
+i8size='1'
+i8type='char'
+i_arpainet='define'
+i_bsdioctl=''
+i_crypt='define'
+i_db='undef'
+i_dbm='undef'
+i_dirent='define'
+i_dld='undef'
+i_dlfcn='define'
+i_fcntl='undef'
+i_float='define'
+i_fp='undef'
+i_fp_class='undef'
+i_gdbm='undef'
+i_grp='define'
+i_ieeefp='undef'
+i_inttypes='define'
+i_langinfo='define'
+i_libutil='undef'
+i_limits='define'
+i_locale='define'
+i_machcthr='undef'
+i_malloc='define'
+i_math='define'
+i_memory='undef'
+i_mntent='define'
+i_ndbm='undef'
+i_netdb='define'
+i_neterrno='undef'
+i_netinettcp='define'
+i_niin='define'
+i_poll='define'
+i_prot='undef'
+i_pthread='define'
+i_pwd='define'
+i_rpcsvcdbm='undef'
+i_sfio='undef'
+i_sgtty='undef'
+i_shadow='define'
+i_socks='undef'
+i_stdarg='define'
+i_stddef='define'
+i_stdlib='define'
+i_string='define'
+i_sunmath='undef'
+i_sysaccess='undef'
+i_sysdir='define'
+i_sysfile='define'
+i_sysfilio='undef'
+i_sysin='undef'
+i_sysioctl='define'
+i_syslog='define'
+i_sysmman='define'
+i_sysmode='undef'
+i_sysmount='define'
+i_sysndir='undef'
+i_sysparam='define'
+i_sysresrc='define'
+i_syssecrt='undef'
+i_sysselct='define'
+i_syssockio='undef'
+i_sysstat='define'
+i_sysstatfs='define'
+i_sysstatvfs='define'
+i_systime='define'
+i_systimek='undef'
+i_systimes='define'
+i_systypes='define'
+i_sysuio='define'
+i_sysun='define'
+i_sysutsname='define'
+i_sysvfs='define'
+i_syswait='define'
+i_termio='undef'
+i_termios='define'
+i_time='define'
+i_unistd='define'
+i_ustat='define'
+i_utime='define'
+i_values='define'
+i_varargs='undef'
+i_varhdr='stdarg.h'
+i_vfork='undef'
+ignore_versioned_solibs='y'
+inc_version_list=' '
+inc_version_list_init='0'
+incpath=''
+inews=''
+installarchlib='/usr/lib/perl5/5.8.8/i686-linux'
+installbin='/usr/bin'
+installhtml1dir=''
+installhtml3dir=''
+installman1dir='/usr/share/man/man1'
+installman3dir='/usr/share/man/man3'
+installprefix='/usr'
+installprefixexp='/usr'
+installprivlib='/usr/lib/perl5/5.8.8'
+installscript='/usr/bin'
+installsitearch='/usr/lib/perl5/site_perl/5.8.8/i686-linux'
+installsitebin='/usr/bin'
+installsitehtml1dir=''
+installsitehtml3dir=''
+installsitelib='/usr/lib/perl5/site_perl/5.8.8'
+installsiteman1dir='/usr/share/man/man1'
+installsiteman3dir='/usr/share/man/man3'
+installsitescript='/usr/bin'
+installstyle='lib/perl5'
+installusrbinperl='undef'
+installvendorarch=''
+installvendorbin=''
+installvendorhtml1dir=''
+installvendorhtml3dir=''
+installvendorlib=''
+installvendorman1dir=''
+installvendorman3dir=''
+installvendorscript=''
+intsize='4'
+issymlink='/bin/test -h'
+ivdformat='"ld"'
+ivsize='4'
+ivtype='long'
+known_extensions='B ByteLoader Cwd DB_File Data/Dumper Devel/DProf Devel/PPPort Devel/Peek Digest/MD5 Encode Fcntl File/Glob Filter/Util/Call GDBM_File I18N/Langinfo IO IPC/SysV List/Util MIME/Base64 NDBM_File ODBM_File Opcode POSIX PerlIO/encoding PerlIO/scalar PerlIO/via SDBM_File Socket Storable Sys/Hostname Sys/Syslog Thread Time/HiRes Unicode/Normalize XS/APItest XS/Typemap attrs re threads threads/shared'
+ksh=''
+ld='cc'
+lddlflags='-shared -L/usr/local/lib'
+ldflags=' -L/usr/local/lib'
+ldflags_uselargefiles=''
+ldlibpthname='LD_LIBRARY_PATH'
+less='less'
+lib_ext='.a'
+libc='/lib/libc-2.4.so'
+libperl='libperl.a'
+libpth='/usr/local/lib /lib /usr/lib'
+libs='-lnsl -ldl -lm -lcrypt -lutil -lc'
+libsdirs=' /usr/lib'
+libsfiles=' libnsl.so libdl.so libm.so libcrypt.so libutil.so libc.so'
+libsfound=' /usr/lib/libnsl.so /usr/lib/libdl.so /usr/lib/libm.so /usr/lib/libcrypt.so /usr/lib/libutil.so /usr/lib/libc.so'
+libspath=' /usr/local/lib /lib /usr/lib'
+libswanted='sfio socket inet nsl nm ndbm gdbm dbm db malloc dl dld ld sun m crypt sec util c cposix posix ucb BSD'
+libswanted_uselargefiles=''
+line=''
+lint=''
+lkflags=''
+ln='ln'
+lns='/bin/ln -s'
+localtime_r_proto='0'
+locincpth='/usr/local/include /opt/local/include /usr/gnu/include /opt/gnu/include /usr/GNU/include /opt/GNU/include'
+loclibpth='/usr/local/lib /opt/local/lib /usr/gnu/lib /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib'
+longdblsize='12'
+longlongsize='8'
+longsize='4'
+lp=''
+lpr=''
+ls='ls'
+lseeksize='8'
+lseektype='off_t'
+mail=''
+mailx=''
+make='make'
+make_set_make='#'
+mallocobj=''
+mallocsrc=''
+malloctype='void *'
+man1dir='/usr/share/man/man1'
+man1direxp='/usr/share/man/man1'
+man1ext='1'
+man3dir='/usr/share/man/man3'
+man3direxp='/usr/share/man/man3'
+man3ext='3'
+mips_type=''
+mistrustnm=''
+mkdir='mkdir'
+mmaptype='void *'
+modetype='mode_t'
+more='more'
+multiarch='undef'
+mv=''
+myarchname='i686-linux'
+mydomain='.jenacon.net'
+myhostname='lfslivecd'
+myuname='linux lfslivecd 2.6.16.14 #1 smp sat may 13 11:48:14 gmt 2006 i686 pentium3 i386 gnulinux '
+n='-n'
+need_va_copy='undef'
+netdb_hlen_type='size_t'
+netdb_host_type='const void *'
+netdb_name_type='const char *'
+netdb_net_type='in_addr_t'
+nm='nm'
+nm_opt=''
+nm_so_opt='--dynamic'
+nonxs_ext='Errno'
+nroff='nroff'
+nvEUformat='"E"'
+nvFUformat='"F"'
+nvGUformat='"G"'
+nv_preserves_uv_bits='32'
+nveformat='"e"'
+nvfformat='"f"'
+nvgformat='"g"'
+nvsize='8'
+nvtype='double'
+o_nonblock='O_NONBLOCK'
+obj_ext='.o'
+old_pthread_create_joinable=''
+optimize='-O2'
+orderlib='false'
+osname='linux'
+osvers='2.6.16.14'
+otherlibdirs=' '
+package='perl5'
+pager='/usr/bin/less -isR'
+passcat='cat /etc/passwd'
+patchlevel='8'
+path_sep=':'
+perl5=''
+perl=''
+perl_patchlevel=''
+perladmin='root@lfslivecd.jenacon.net'
+perllibs='-lnsl -ldl -lm -lcrypt -lutil -lc'
+perlpath='/usr/bin/perl'
+pg='pg'
+phostname='hostname'
+pidtype='pid_t'
+plibpth=''
+pmake=''
+pr=''
+prefix='/usr'
+prefixexp='/usr'
+privlib='/usr/lib/perl5/5.8.8'
+privlibexp='/usr/lib/perl5/5.8.8'
+procselfexe='"/proc/self/exe"'
+prototype='define'
+ptrsize='4'
+quadkind='3'
+quadtype='long long'
+randbits='48'
+randfunc='drand48'
+random_r_proto='0'
+randseedtype='long'
+ranlib=':'
+rd_nodata='-1'
+readdir64_r_proto='0'
+readdir_r_proto='0'
+revision='5'
+rm='rm'
+rmail=''
+run=''
+runnm='false'
+sPRIEUldbl='"LE"'
+sPRIFUldbl='"LF"'
+sPRIGUldbl='"LG"'
+sPRIXU64='"LX"'
+sPRId64='"Ld"'
+sPRIeldbl='"Le"'
+sPRIfldbl='"Lf"'
+sPRIgldbl='"Lg"'
+sPRIi64='"Li"'
+sPRIo64='"Lo"'
+sPRIu64='"Lu"'
+sPRIx64='"Lx"'
+sSCNfldbl='"Lf"'
+sched_yield='sched_yield()'
+scriptdir='/usr/bin'
+scriptdirexp='/usr/bin'
+sed='sed'
+seedfunc='srand48'
+selectminbits='32'
+selecttype='fd_set *'
+sendmail=''
+setgrent_r_proto='0'
+sethostent_r_proto='0'
+setlocale_r_proto='0'
+setnetent_r_proto='0'
+setprotoent_r_proto='0'
+setpwent_r_proto='0'
+setservent_r_proto='0'
+sh='/bin/sh'
+shar=''
+sharpbang='#!'
+shmattype='void *'
+shortsize='2'
+shrpenv=''
+shsharp='true'
+sig_count='65'
+sig_name='ZERO HUP INT QUIT ILL TRAP ABRT BUS FPE KILL USR1 SEGV USR2 PIPE ALRM TERM STKFLT CHLD CONT STOP TSTP TTIN TTOU URG XCPU XFSZ VTALRM PROF WINCH IO PWR SYS NUM32 NUM33 RTMIN NUM35 NUM36 NUM37 NUM38 NUM39 NUM40 NUM41 NUM42 NUM43 NUM44 NUM45 NUM46 NUM47 NUM48 NUM49 NUM50 NUM51 NUM52 NUM53 NUM54 NUM55 NUM56 NUM57 NUM58 NUM59 NUM60 NUM61 NUM62 NUM63 RTMAX IOT CLD POLL UNUSED '
+sig_name_init='"ZERO", "HUP", "INT", "QUIT", "ILL", "TRAP", "ABRT", "BUS", "FPE", "KILL", "USR1", "SEGV", "USR2", "PIPE", "ALRM", "TERM", "STKFLT", "CHLD", "CONT", "STOP", "TSTP", "TTIN", "TTOU", "URG", "XCPU", "XFSZ", "VTALRM", "PROF", "WINCH", "IO", "PWR", "SYS", "NUM32", "NUM33", "RTMIN", "NUM35", "NUM36", "NUM37", "NUM38", "NUM39", "NUM40", "NUM41", "NUM42", "NUM43", "NUM44", "NUM45", "NUM46", "NUM47", "NUM48", "NUM49", "NUM50", "NUM51", "NUM52", "NUM53", "NUM54", "NUM55", "NUM56", "NUM57", "NUM58", "NUM59", "NUM60", "NUM61", "NUM62", "NUM63", "RTMAX", "IOT", "CLD", "POLL", "UNUSED", 0'
+sig_num='0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 6 17 29 31 '
+sig_num_init='0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 6, 17, 29, 31, 0'
+sig_size='69'
+signal_t='void'
+sitearch='/usr/lib/perl5/site_perl/5.8.8/i686-linux'
+sitearchexp='/usr/lib/perl5/site_perl/5.8.8/i686-linux'
+sitebin='/usr/bin'
+sitebinexp='/usr/bin'
+sitehtml1dir=''
+sitehtml1direxp=''
+sitehtml3dir=''
+sitehtml3direxp=''
+sitelib='/usr/lib/perl5/site_perl/5.8.8'
+sitelib_stem='/usr/lib/perl5/site_perl'
+sitelibexp='/usr/lib/perl5/site_perl/5.8.8'
+siteman1dir='/usr/share/man/man1'
+siteman1direxp='/usr/share/man/man1'
+siteman3dir='/usr/share/man/man3'
+siteman3direxp='/usr/share/man/man3'
+siteprefix='/usr'
+siteprefixexp='/usr'
+sitescript='/usr/bin'
+sitescriptexp='/usr/bin'
+sizesize='4'
+sizetype='size_t'
+sleep=''
+smail=''
+so='so'
+sockethdr=''
+socketlib=''
+socksizetype='socklen_t'
+sort='sort'
+spackage='Perl5'
+spitshell='cat'
+srand48_r_proto='0'
+srandom_r_proto='0'
+src='.'
+ssizetype='ssize_t'
+startperl='#!/usr/bin/perl'
+startsh='#!/bin/sh'
+static_ext=' '
+stdchar='char'
+stdio_base='((fp)->_IO_read_base)'
+stdio_bufsiz='((fp)->_IO_read_end - (fp)->_IO_read_base)'
+stdio_cnt='((fp)->_IO_read_end - (fp)->_IO_read_ptr)'
+stdio_filbuf=''
+stdio_ptr='((fp)->_IO_read_ptr)'
+stdio_stream_array=''
+strerror_r_proto='0'
+strings='/usr/include/string.h'
+submit=''
+subversion='8'
+sysman='/usr/share/man/man1'
+tail=''
+tar=''
+targetarch=''
+tbl=''
+tee=''
+test='test'
+timeincl='/usr/include/sys/time.h /usr/include/time.h '
+timetype='time_t'
+tmpnam_r_proto='0'
+to=':'
+touch='touch'
+tr='tr'
+trnl='\n'
+troff=''
+ttyname_r_proto='0'
+u16size='2'
+u16type='unsigned short'
+u32size='4'
+u32type='unsigned long'
+u64size='8'
+u64type='unsigned long long'
+u8size='1'
+u8type='unsigned char'
+uidformat='"lu"'
+uidsign='1'
+uidsize='4'
+uidtype='uid_t'
+uname='uname'
+uniq='uniq'
+uquadtype='unsigned long long'
+use5005threads='undef'
+use64bitall='undef'
+use64bitint='undef'
+usecrosscompile='undef'
+usedl='define'
+usefaststdio='define'
+useithreads='undef'
+uselargefiles='define'
+uselongdouble='undef'
+usemallocwrap='define'
+usemorebits='undef'
+usemultiplicity='undef'
+usemymalloc='n'
+usenm='false'
+useopcode='true'
+useperlio='define'
+useposix='true'
+usereentrant='undef'
+usesfio='false'
+useshrplib='false'
+usesitecustomize='undef'
+usesocks='undef'
+usethreads='undef'
+usevendorprefix='undef'
+usevfork='false'
+usrinc='/usr/include'
+uuname=''
+uvXUformat='"lX"'
+uvoformat='"lo"'
+uvsize='4'
+uvtype='unsigned long'
+uvuformat='"lu"'
+uvxformat='"lx"'
+vendorarch=''
+vendorarchexp=''
+vendorbin=''
+vendorbinexp=''
+vendorhtml1dir=' '
+vendorhtml1direxp=''
+vendorhtml3dir=' '
+vendorhtml3direxp=''
+vendorlib=''
+vendorlib_stem=''
+vendorlibexp=''
+vendorman1dir=' '
+vendorman1direxp=''
+vendorman3dir=' '
+vendorman3direxp=''
+vendorprefix=''
+vendorprefixexp=''
+vendorscript=''
+vendorscriptexp=''
+version='5.8.8'
+version_patchlevel_string='version 8 subversion 8'
+versiononly='undef'
+vi=''
+voidflags='15'
+xlibpth='/usr/lib/386 /lib/386'
+yacc='yacc'
+yaccflags=''
+zcat=''
+zip='zip'
+# Configure command line arguments.
+config_arg0='Configure'
+config_args='-ds -e -Dprefix=/usr -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dpager=/usr/bin/less -isR'
+config_argc=6
+config_arg1='-ds'
+config_arg2='-e'
+config_arg3='-Dprefix=/usr'
+config_arg4='-Dman1dir=/usr/share/man/man1'
+config_arg5='-Dman3dir=/usr/share/man/man3'
+config_arg6='-Dpager=/usr/bin/less -isR'
+PERL_REVISION=5
+PERL_VERSION=8
+PERL_SUBVERSION=8
+PERL_API_REVISION=5
+PERL_API_VERSION=8
+PERL_API_SUBVERSION=0
+PERL_PATCHLEVEL=
+PERL_CONFIG_SH=true
diff -Naur perl-5.8.8.orig/Cross/generate_config_sh perl-5.8.8/Cross/generate_config_sh
--- perl-5.8.8.orig/Cross/generate_config_sh	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/generate_config_sh	1969-12-31 19:00:00.000000000 -0500
@@ -1,132 +0,0 @@
-#!/usr/bin/perl
-
-##############################################################################
-#
-#	generate_config_sh
-#		Process that takes an automatically generated config.sh
-#		file and allows the environment to overload the values
-#		automatically discovered by Configure on our target platform.
-#
-#	Author	Redvers Davies <red@criticalintegration.com>
-#
-##############################################################################
-
-my $config = shift;
-
-my $sys = $ENV{SYS};
-
-my $callbacks = {};
-$callbacks->{'ar'} = [\&simple_process, ["AR", "arm-linux-ar"]];
-$callbacks->{'archname'} = [\&simple_process, ["SYS", "armv4l-linux"]];
-$callbacks->{'cc'} = [\&simple_process, ["CC", "arm-linux-gcc"]];
-$callbacks->{'cccdlflags'} = [\&simple_process, ["CFLAGS", ""]];
-$callbacks->{'ccdlflags'} = [\&simple_process, ["CFLAGS", ""]];
-$callbacks->{'ccflags'} = [\&simple_process, ["CFLAGS", "-fno-strict-aliasing -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"]];
-$callbacks->{'ccflags_uselargefiles'} = [\&simple_process, ["CFLAGS", "-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"]];
-$callbacks->{'ccname'} = [\&simple_process, ["CC", "arm-linux-gcc"]];
-$callbacks->{'cpp'} = [\&simple_process, ["CCP", "arm-linux-cpp"]];
-$callbacks->{'cppflags'} = [\&simple_process, ["CCPFLAGS", "-fno-strict-aliasing"]];
-$callbacks->{'cpprun'} = [\&simple_process_append, ["CC", "arm-linux-gcc -E", "-E"]];
-$callbacks->{'cppstdin'} = [\&simple_process_append, ["CC", "arm-linux-gcc -E", "-E"]];
-$callbacks->{'full_ar'} = [\&backtick, ["AR", "which $ENV{AR}", "/usr/local/arm/2.95.3/bin/arm-linux-ar"]];
-$callbacks->{'ld'} = [\&simple_process, ["LD", "arm-linux-ld"]];
-$callbacks->{'ldflags'} = [\&simple_process, ["LDFLAGS", ""]];
-$callbacks->{'ldflags_uselargefiles'} = [\&simple_process, ["LDFLAGS", ""]];
-$callbacks->{'myarchname'} = [\&simple_process, ["SYS", "armv4l-linux"]];
-$callbacks->{'archlib'} = [\&library_munge, ["SYS", "armv4l-linux"]];
-$callbacks->{'archlibexp'} = [\&library_munge, ["SYS", "armv4l-linux"]];
-$callbacks->{'installarchlib'} = [\&library_munge, ["SYS", "armv4l-linux"]];
-$callbacks->{'installsitearch'} = [\&library_munge, ["SYS", "armv4l-linux"]];
-$callbacks->{'sitearch'} = [\&library_munge, ["SYS", "armv4l-linux"]];
-$callbacks->{'sitearchexp'} = [\&library_munge, ["SYS", "armv4l-linux"]];
-
-if ($config eq "") {
-	die("Please run me as generate_config_sh path/to/original/config.sh");
-}
-
-open(FILE, "$config") || die("Unable to open $config");
-
-my $line_in;
-while ($line_in = <FILE>) {
-	if ($line_in =~ /^#/) {
-		next;
-	}
-
-	if ($line_in !~ /./) {
-		next;
-	}
-
-	$line_in =~ /^([^=]+)=(.*)/;
-	my $key = $1;
-	my $value = $2;
-	if (ref($callbacks->{$key}) eq "ARRAY") {
-		($callbacks->{$key}[0])->($key,$value);
-	} else {
-		print($line_in);
-	}
-}
-
-sub backtick {
-	my $key = shift;
-	my $value = shift;
-	my $envvar = $callbacks->{$key}->[1][0];
-
-	if ($ENV{$envvar}) {
-		my $rawtext = `$callbacks->{$key}->[1][1]`;
-		chomp($rawtext);
-		print("$key=\'$rawtext\'\n");
-	} else {
-		print("$key=\'$callbacks->{$key}->[1][2]\'\n");
-	}
-}
-	
-
-sub simple_process {
-	my $key = shift;
-	my $envvar = $callbacks->{$key}->[1][0];
-
-	if ($ENV{$envvar}) {
-		print("$key=\"$ENV{$envvar}\"\n");
-	} else {
-		print("$key=\'$callbacks->{$key}->[1][1]\'\n");
-	}
-
-}
-
-sub simple_process_append {
-	my $key = shift;
-	my $envvar = $callbacks->{$key}->[1][0];
-
-	if ($ENV{$envvar}) {
-		print("$key=\"$ENV{$envvar} $callbacks->{$key}->[1][2]\"\n");
-	} else {
-		print("$key=\'$callbacks->{$key}->[1][1]\'\n");
-	}
-
-}
-
-sub library_munge {
-	my $key = shift;
-	my $value = shift;
-	my $envvar = $callbacks->{$key}->[1][0];
-
-	if ($ENV{$envvar}) {
-		$value =~ s/$callbacks->{$key}->[1][1]/$ENV{$envvar}/g;
-		print("$key=$value\n");
-	} else {
-		print("$key=$value\n");
-	}
-		
-}
-
-
-
-
-
-
-
-
-
-
-
-
diff -Naur perl-5.8.8.orig/Cross/installperl.patch perl-5.8.8/Cross/installperl.patch
--- perl-5.8.8.orig/Cross/installperl.patch	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/installperl.patch	1969-12-31 19:00:00.000000000 -0500
@@ -1,22 +0,0 @@
---- ../installperl	2003-08-31 11:58:48.000000000 -0500
-+++ installperl	2003-09-05 02:38:40.000000000 -0500
-@@ -3,8 +3,8 @@
- BEGIN {
-     require 5.004;
-     chdir '..' if !-d 'lib' and -d '../lib';
--    @INC = 'lib';
--    $ENV{PERL5LIB} = 'lib';
-+#    @INC = 'lib';
-+#    $ENV{PERL5LIB} = 'lib';
- }
- 
- use strict;
-@@ -946,7 +946,7 @@
- 		print " " if (@opts);
- 		print "$file\n";
- 	    }
--	    system("strip", @opts, $file);
-+	    system("XXSTRIPXX-strip", @opts, $file);
- 	} else {
- 	    print "# file '$file' skipped\n" if $verbose;
- 	}
diff -Naur perl-5.8.8.orig/Cross/Makefile perl-5.8.8/Cross/Makefile
--- perl-5.8.8.orig/Cross/Makefile	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/Makefile	2006-07-22 18:17:35.000000000 -0400
@@ -1,68 +1,62 @@
-## Toplevel Makefile for cross-compilation of perl
-#
-## $Id: Makefile,v 1.7 2004/01/12 15:41:02 red Exp red $
-
-export TOPDIR=${shell pwd}
-include $(TOPDIR)/config
-export CFLAGS
-export SYS=$(ARCH)-$(OS)
-export CROSS=$(ARCH)-$(OS)-
-export FULL_OPTIMIZATION = -fexpensive-optimizations -fomit-frame-pointer -O2
-export OPTIMIZATION = -O2
-
-export CC = $(CROSS)gcc
-export CXX = $(CROSS)g++
-export LD = $(CROSS)ld
-export STRIP = $(CROSS)strip
-export AR = $(CROSS)ar
-export RANLIB = $(CROSS)ranlib
-
-
-## Optimisation work
-ifeq ($(ARCH),arm)
- ifdef CONFIG_TARGET_ARM_SA11X0
-   ifndef Architecture
-     Architecture = armv4l-strongarm
-   endif
-   FULL_OPTIMIZATION += -march=armv4 -mtune=strongarm1100 -mapcs-32
-   OPTIMIZATION += -march=armv4 -mtune=strongarm1100 -mapcs-32
- endif
-endif
+PWD	:= $(shell pwd)
+TOP	:= $(shell dirname $(PWD))
 
-CFLAGS+=$(FULL_OPTIMIZATION)
+VERSION	:= $(shell awk '/define[ ]PERL_REVISION/ {printf $$3"."} /define[ ]PERL_VERSION/ {printf $$3"."} /define[ ]PERL_SUBVERSION/ {print $$3}' $(TOP)/patchlevel.h)
 
-all:
-	@echo Please read the README file before doing anything else.
+CROSS_COMPILE := arm-unknown-linux-gnu-
+
+ARCH	:= arm
+OS	:= linux
+
+SYS	:= $(ARCH)-$(OS)
 
-gen_patch:
-	diff -Bbur ../Makefile.SH Makefile.SH > Makefile.SH.patch
-	diff -Bbur ../installperl installperl > installperl.patch
-
-patch:
-	cd .. ; if [ ! -e ./CROSS_PATCHED ] ; then \
-		patch -p1 < Cross/Makefile.SH.patch; \
-		patch -p1 < Cross/installperl.patch ; mv installperl installperl-patched; \
-		sed -e 's/XXSTRIPXX/$(SYS)/' installperl-patched > installperl; \
-		touch CROSS_PATCHED ; fi
-
-dry_patch:
-	cd .. ;	patch --dry-run -p1 < Cross/Makefile.SH.patch; \
-		patch --dry-run -p1 < Cross/installperl.patch; \
-
-perl:
-	@echo Perl cross-build directory is $(TOPDIR)
-	@echo Target arch is $(SYS)
-	@echo toolchain: $(CC), $(CXX), $(LD), $(STRIP), $(AR), $(RANLIB)
-	@echo Optimizations: $(FULL_OPTIMIZATION)
-
-	$(TOPDIR)/generate_config_sh config.sh-$(SYS) > $(TOPDIR)/../config.sh
-	cd $(TOPDIR)/.. ; ./Configure -S ; make depend ; make ; make more
-	cd $(TOPDIR)/.. ; mkdir -p fake_config_library ; cp lib/Config.pm fake_config_library
-	cd $(TOPDIR)/.. ; $(MAKE) more2 "PERLRUN=/usr/bin/perl -I$(TOPDIR)/../fake_config_library -MConfig"
-	cd $(TOPDIR)/.. ; $(MAKE) more3 "PERLRUN=/usr/bin/perl -I$(TOPDIR)/../fake_config_library -MConfig"
-	cd $(TOPDIR)/.. ; $(MAKE) more4 "PERLRUN=/usr/bin/perl -I$(TOPDIR)/../fake_config_library -MConfig"
-	cd $(TOPDIR)/.. ; rm -rf install_me_here
-	cd $(TOPDIR)/.. ; make install-strip
-	cd $(TOPDIR)/.. ; sh -x Cross/warp
+AR	:= $(CROSS_COMPILE)ar
+CC	:= $(CROSS_COMPILE)gcc
+CXX	:= $(CROSS_COMPILE)g++
+CPP	:= $(CROSS_COMPILE)cpp
+LD	:= $(CROSS_COMPILE)ld
+RANLIB	:= $(CROSS_COMPILE)ranlib
+STRIP	:= $(CROSS_COMPILE)strip
 
+.PHONY: all install create-config.sh
+
+all:
+	cd $(TOP); CC=gcc ./configure.gnu
 
+	# Make miniperl
+	$(MAKE) -C $(TOP) miniperl
+	mv -v $(TOP)/miniperl $(TOP)/miniperl-cross
+	$(MAKE) -C $(TOP) clean
+
+	# Do the final build
+	$(MAKE) create-config.sh
+	cd $(TOP); ./Configure -S
+	$(MAKE) -C $(TOP)
+
+install:
+	$(MAKE) -C $(TOP) install
+
+create-config.sh:
+	sed -e "s@\(^archname=\).*@\1'$(SYS)'@g" \
+	    -e "s@\(^myarchname=\).*@\1'$(SYS)'@g" \
+	    -e "s@\(^ar=\).*@\1'$(AR)'@g" \
+	    -e "s@\(^cc=\).*@\1'$(CC)'@g" \
+	    -e "s@\(^ccname=\).*@\1'$(CC)'@g" \
+	    -e "s@\(^cpp=\).*@\1'$(CPP)'@g" \
+	    -e "s@\(^cpprun=\).*@\1'$(CC) -E'@g" \
+	    -e "s@\(^cppstdin=\).*@\1'$(CC) -E'@g" \
+	    -e "s@\(^ld=\).*@\1'$(LD)'@g" \
+	    -e "s@\(^ranlib=\).*@\1'$(RANLIB)'@g" \
+	    -e "s@\(^full_ar=\).*@\1'$(shell which $(AR))'@g" \
+	    -e "s@\(^full_sed=\).*@\1'$(shell which sed)'@g" \
+	    -e "/^ccflags=/s@-I/usr/local/include@@g" \
+	    -e "/^cppflags=/s@-I/usr/local/include@@g" \
+	    -e "/^ldflags=/s@-L/usr/local/lib@@g" \
+	    -e "/^lddlflags=/s@-L/usr/local/lib@@g" \
+	    -e "/^archlib=/s@$(VERSION).*@$(VERSION)/$(SYS)'@" \
+	    -e "/^archlibexp=/s@$(VERSION).*@$(VERSION)/$(SYS)'@" \
+	    -e "/^installarchlib=/s@$(VERSION).*@$(VERSION)/$(SYS)'@" \
+	    -e "/^installsitearch=/s@$(VERSION).*@$(VERSION)/$(SYS)'@" \
+	    -e "/^sitearch/s@$(VERSION).*@$(VERSION)/$(SYS)'@" \
+	    -e "/^sitearchexp/s@$(VERSION).*@$(VERSION)/$(SYS)'@" \
+	    $(PWD)/config.sh-$(SYS) > $(TOP)/config.sh
diff -Naur perl-5.8.8.orig/Cross/Makefile.SH.patch perl-5.8.8/Cross/Makefile.SH.patch
--- perl-5.8.8.orig/Cross/Makefile.SH.patch	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/Makefile.SH.patch	1969-12-31 19:00:00.000000000 -0500
@@ -1,189 +0,0 @@
---- ../Makefile.SH
-+++ ../Makefile.SH
-@@ -129,18 +129,7 @@
- 	    # INSTALL file, under "Building a shared perl library".
- 	    # If there is no pre-existing $libperl, we don't need
- 	    # to do anything further.
--	    if test -f $archlib/CORE/$libperl; then
--		rm -f preload
--		cat <<'EOT' > preload
--#! /bin/sh
--lib=$1
--shift
--test -r $lib && export LD_PRELOAD="$lib $LD_PRELOAD"
--exec "$@"
--EOT
--		chmod 755 preload
--		ldlibpth="$ldlibpth `pwd`/preload `pwd`/$libperl"
--	    fi
-+	    echo linux libraries overwritten by cross-compile patches
- 	    ;;
- 	os390)	test -f /bin/env && ldlibpth="/bin/env $ldlibpth"
- 		;;
-@@ -389,9 +378,21 @@
- .c.s:
- 	$(CCCMDSRC) -S $*.c
- 
--all: $(FIRSTMAKEFILE) miniperl$(EXE_EXT) extra.pods $(private) $(unidatafiles) $(public) $(dynamic_ext) $(nonxs_ext) extras.make
--	@echo " ";
--	@echo "	Everything is up to date. Type '$(MAKE) test' to run test suite."
-+#all: $(FIRSTMAKEFILE) miniperl$(EXE_EXT) extra.pods $(private) $(unidatafiles) $(public) $(dynamic_ext) $(nonxs_ext) extras.make
-+#	@echo " ";
-+#	@echo "	Everything is up to date. Type '$(MAKE) test' to run test suite."
-+
-+all: $(FIRSTMAKEFILE) miniperl$(EXE_EXT)
-+	mv miniperl miniperl-arm
-+	ln -s /usr/bin/perl miniperl
-+
-+more: extra.pods $(private) $(public)
-+
-+more2: $(dynamic_ext)
-+
-+more3: $(nonxs_ext)
-+
-+more4: extras.make
- 
- .PHONY: all compile translators utilities
- 
-@@ -401,10 +402,10 @@
- 	cd x2p; $(MAKE) compile;
- 	cd pod; $(MAKE) compile;
- 
--translators:	miniperl$(EXE_EXT) $(CONFIGPM) FORCE
-+translators:	$(CONFIGPM) FORCE
- 	@echo " "; echo "	Making x2p stuff"; cd x2p; $(LDLIBPTH) $(MAKE) all
- 
--utilities:	miniperl$(EXE_EXT) $(CONFIGPM) $(plextract) lib/lib.pm FORCE
-+utilities:	$(CONFIGPM) $(plextract) lib/lib.pm FORCE
- 	@echo " "; echo "	Making utilities"; cd utils; $(LDLIBPTH) $(MAKE) all
- 
- 
-@@ -579,7 +580,7 @@
- 	$(CC) -o miniperl $(CLDFLAGS) \
- 	    `echo $(obj) | sed 's/ op$(OBJ_EXT) / /'` \
- 	    miniperlmain$(OBJ_EXT) opmini$(OBJ_EXT) perl$(OBJ_EXT) $(libs)
--	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
-+#	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
- !NO!SUBS!
- 		;;
- 	next4*)
-@@ -587,7 +588,7 @@
- miniperl: $& miniperlmain$(OBJ_EXT) $(LIBPERL) opmini$(OBJ_EXT)
- 	$(CC) -o miniperl `echo $(obj) | sed 's/ op$(OBJ_EXT) / /'` \
- 	    miniperlmain$(OBJ_EXT) opmini$(OBJ_EXT) perl$(OBJ_EXT) $(libs)
--	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
-+#	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
- !NO!SUBS!
- 		;;
- 	darwin*)
-@@ -754,7 +755,7 @@
- # We need to autosplit in two steps because VOS can't handle so many args
- #
- .PHONY: preplibrary
--preplibrary: miniperl$(EXE_EXT) $(CONFIGPM) lib/lib.pm $(PREPLIBRARY_LIBPERL)
-+preplibrary: $(CONFIGPM) lib/lib.pm $(PREPLIBRARY_LIBPERL)
- 	@sh ./makedir lib/auto
- 	@echo "	AutoSplitting perl library"
- 	$(LDLIBPTH) ./miniperl -Ilib -e 'use AutoSplit; \
-@@ -763,35 +764,35 @@
- 		autosplit_lib_modules(@ARGV)' lib/*/*.pm
- 	$(MAKE) lib/re.pm
- 
--lib/Config.pod: config.sh miniperl$(EXE_EXT) configpm Porting/Glossary
-+lib/Config.pod: config.sh configpm Porting/Glossary
- 	$(LDLIBPTH) ./miniperl -Ilib configpm --heavy=lib/Config_heavy.pl lib/Config.pm
- 
- $(CONFIGPM): lib/Config.pod
- 
--lib/ExtUtils/Miniperl.pm: miniperlmain.c miniperl$(EXE_EXT) minimod.pl $(CONFIGPM)
-+lib/ExtUtils/Miniperl.pm: miniperlmain.c minimod.pl $(CONFIGPM)
- 	$(LDLIBPTH) ./miniperl minimod.pl > lib/ExtUtils/Miniperl.pm
- 
- lib/re.pm: ext/re/re.pm
- 	cp ext/re/re.pm lib/re.pm
- 
--$(plextract):	miniperl$(EXE_EXT) $(CONFIGPM) x2p/s2p
-+$(plextract):	$(CONFIGPM) x2p/s2p
- 	@-rm -f $@
- 	$(LDLIBPTH) ./miniperl -I`pwd`/lib $@.PL
- 
--x2p/s2p: miniperl$(EXE_EXT) $(CONFIGPM) x2p/s2p.PL
-+x2p/s2p: $(CONFIGPM) x2p/s2p.PL
- 	cd x2p; $(LDLIBPTH) $(MAKE) s2p
- 
--lib/lib.pm:	miniperl$(EXE_EXT) $(CONFIGPM)
-+lib/lib.pm:	$(CONFIGPM)
- 	@-rm -f $@
- 	$(LDLIBPTH) ./miniperl -Ilib lib/lib_pm.PL
- 
- unidatafiles $(unidatafiles): uni.data
- 
--uni.data: miniperl$(EXE_EXT) $(CONFIGPM) lib/unicore/mktables
-+uni.data: $(CONFIGPM) lib/unicore/mktables
- 	cd lib/unicore && $(LDLIBPTH) ../../miniperl -I../../lib mktables -w
- 	touch uni.data
- 
--extra.pods: miniperl$(EXE_EXT)
-+extra.pods:
- 	-@test -f extra.pods && rm -f `cat extra.pods`
- 	-@rm -f extra.pods
- 	-@for x in `grep -l '^=[a-z]' README.* | grep -v README.vms` ; do \
-@@ -798,18 +799,7 @@
- INSTALL_DEPENDENCE = all
- 
- install.perl:	$(INSTALL_DEPENDENCE) installperl
--	if [ -n "$(COMPILE)" ]; \
--	then \
--		cd utils; $(MAKE) compile; \
--		cd ../x2p; $(MAKE) compile; \
--		cd ../pod; $(MAKE) compile; \
--	else :; \
--	fi
--	$(LDLIBPTH) ./perl installperl --destdir=$(DESTDIR) $(INSTALLFLAGS) $(STRIPFLAGS)
--	$(MAKE) extras.install
--
--install.man:	all installman
--	$(LDLIBPTH) ./perl installman --destdir=$(DESTDIR) $(INSTALLFLAGS)
-+	/usr/bin/perl -Ifake_config_library -MConfig installperl $(INSTALLFLAGS) $(STRIPFLAGS)
- 
- # XXX Experimental. Hardwired values, but useful for testing.
- # Eventually Configure could ask for some of these values.
-@@ -925,16 +915,16 @@
- #
- # DynaLoader may be needed for extensions that use Makefile.PL.
- 
--$(DYNALOADER):	miniperl$(EXE_EXT) preplibrary FORCE
-+$(DYNALOADER):	preplibrary FORCE
- 	@$(LDLIBPTH) sh ext/util/make_ext $(STATIC) $@ MAKE=$(MAKE) LIBPERL_A=$(LIBPERL)
- 
--d_dummy $(dynamic_ext):	miniperl$(EXE_EXT) preplibrary $(DYNALOADER) FORCE
-+d_dummy $(dynamic_ext):	preplibrary $(DYNALOADER) FORCE
- 	@$(LDLIBPTH) sh ext/util/make_ext dynamic $@ MAKE=$(MAKE) LIBPERL_A=$(LIBPERL)
- 
--s_dummy $(static_ext):	miniperl$(EXE_EXT) preplibrary $(DYNALOADER) FORCE
-+s_dummy $(static_ext):	preplibrary $(DYNALOADER) FORCE
- 	@$(LDLIBPTH) sh ext/util/make_ext $(STATIC) $@ MAKE=$(MAKE) LIBPERL_A=$(LIBPERL)
- 
--n_dummy $(nonxs_ext):	miniperl$(EXE_EXT) preplibrary $(DYNALOADER) FORCE
-+n_dummy $(nonxs_ext):	preplibrary $(DYNALOADER) FORCE
- 	@$(LDLIBPTH) sh ext/util/make_ext nonxs $@ MAKE=$(MAKE) LIBPERL_A=$(LIBPERL)
- 
- .PHONY: clean _tidy _mopup _cleaner1 _cleaner2 \
-@@ -1074,7 +1064,7 @@
- 
- test_prep_pre: preplibrary utilities $(nonxs_ext)
- 
--test_prep: test_prep_pre miniperl$(EXE_EXT) $(unidatafiles) perl$(EXE_EXT) $(dynamic_ext) $(TEST_PERL_DLL)
-+test_prep: test_prep_pre $(unidatafiles) perl$(EXE_EXT) $(dynamic_ext) $(TEST_PERL_DLL)
- 	PERL=./perl $(MAKE) _test_prep
- 
- _test_tty:
-@@ -1167,7 +1157,7 @@
- 
- # Can't depend on lib/Config.pm because that might be where miniperl
- # is crashing.
--minitest: miniperl$(EXE_EXT) lib/re.pm minitest.prep
-+minitest: lib/re.pm minitest.prep
- 	- cd t && (rm -f perl$(EXE_EXT); $(LNS) ../miniperl$(EXE_EXT) perl$(EXE_EXT)) \
- 		&& $(LDLIBPTH) ./perl TEST -minitest base/*.t comp/*.t cmd/*.t run/*.t io/*.t op/*.t uni/*.t </dev/tty
- 
diff -Naur perl-5.8.8.orig/Cross/README perl-5.8.8/Cross/README
--- perl-5.8.8.orig/Cross/README	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/README	1969-12-31 19:00:00.000000000 -0500
@@ -1,90 +0,0 @@
-Building for arm-linux
-----------------------
-
-The files in this directory add another cross-compilation
-target to the Perl buildsystem.  It was built as a part of
-the Open Zaurus (http://www.openzaurus.com/) distribution.
-Most / All of the arm compiler optimisations are "borrowed"
-from this excellent project.
-
-(Further discussion about cross-compiling Perl in the top level
-INSTALL file, see the section "Cross-compilation".)
-
-The main target is arm-linux but I have also managed to
-successfully cross-compile Perl for Solaris x86 using the same
-buildsystem.
-
-We are currently dependent on an existing working local copy of
-Perl ** of the same version and revision ** which is available
-as /usr/bin/perl. 
-
-You need a working and tested cross-compiler for your build
-and target combination.  The binary directory must be in
-your path.
-
-1)	You should be reading me (README) in perl-5.x.y/Cross
-
-2)	Make sure you are in the Cross directory.
-
-3)	Edit the file 'config' to contain your target platform information.
-
-4)	make patch	## This will patch the existing source-tree.
-5)	make perl	## Will make perl
-
-Your built Perl environment is in install_me_here/ in your build
-directory.  From here you can package and deploy as you wish.
-
-The Obvious Ommissions
-----------------------
-
-This does NOT perform any installation as site installation method
-will be dependent on the target architecture and OS. 
-
-make test will NOT work as the binaries and libraries will not execute
-on your BUILD machine. 
-
-Due to space limitations on the Zaurus (it's a PDA) we do not provide
-documentation in the core - Therefore man pages are not even generated.
-
-Other Targets (For Developers)
-------------------------------
-
-It is possible to extend the cross-compilation to other targets.
-We have sucessfully compiled for the target solaris2.8/x86
-on linux/x86 build system.
-
-To attempt a cross-compile for another target using the methods
-in this directory:
-
-1)	Copy the Perl source code onto your TARGET machine.
-2)	Execute sh Configure as normal and configure as required,
-	do not "make".
-3)	Copy the config.sh file that is generated to your BUILD
-	machine and place it in the Cross directory with the
-	filename config.sh-ARCH-OS.  For example,
-	config.sh-i386-pc-solaris2.8.  For the appropriate ARCH
-	and OS please refer to your cross-compiler documentation.
-4)	Edit Cross/config to reflect your new target and continue
-	with build as above.
-
-Should you wish to produce optimised binaries for different
-architectures you can add the appropriate compiler flags to
-the Makefile in a new ifeq ($(ARCH),...) ... endif block.
-
-
-Please refer to your cross-compiler documentation for details.
-
-
-	Note that the Cross/ directory is also used by a different
-	cross-compilation setup described in the INSTALL file, and
-	executed by Configure.  There should be no conflicts since
-	it is unlikely both that cross-compilation setups are used
-	simultaneously.
-
-Enjoy!
-
-References
-----------
-Redvers Davies <red@criticalintegration.com>
-Open Zaurus http://www.openzaurus.org/
-Perl OZ Packages http://www.criticalintegration.com/perl-oz/
diff -Naur perl-5.8.8.orig/Cross/TODO perl-5.8.8/Cross/TODO
--- perl-5.8.8.orig/Cross/TODO	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/TODO	1969-12-31 19:00:00.000000000 -0500
@@ -1 +0,0 @@
-Provide a better sandbox for building additional XS libraries.
diff -Naur perl-5.8.8.orig/Cross/warp perl-5.8.8/Cross/warp
--- perl-5.8.8.orig/Cross/warp	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/Cross/warp	1969-12-31 19:00:00.000000000 -0500
@@ -1,14 +0,0 @@
-#!/bin/sh -x
-
-## This script contains an rm -rf.  This may alarm you.
-## These directories need to be reccursively deleted.
-## I don't see any method of making "Bad Things"[tm]
-## Happen.  But we don't run buildroot as root do we? :-)
-
-for f in `find lib -name install_me_here`;do
-cp -r $f/* ./install_me_here/
-done
-
-for f in `find lib -name install_me_here`;do
-rm -rf $f
-done
diff -Naur perl-5.8.8.orig/ext/util/make_ext perl-5.8.8/ext/util/make_ext
--- perl-5.8.8.orig/ext/util/make_ext	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/ext/util/make_ext	2006-07-22 18:17:35.000000000 -0400
@@ -119,7 +119,7 @@
 esac
 
 if test ! -f $makefile ; then
-	test -f Makefile.PL && ../$depth/miniperl -I../$depth/lib Makefile.PL INSTALLDIRS=perl PERL_CORE=1 $passthru
+	test -f Makefile.PL && ../$depth/miniperl-cross -I../$depth/lib Makefile.PL INSTALLDIRS=perl PERL_CORE=1 $passthru
 fi
 if test ! -f $makefile ; then
 	if test -f Makefile.SH; then
diff -Naur perl-5.8.8.orig/installman perl-5.8.8/installman
--- perl-5.8.8.orig/installman	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/installman	2006-07-22 18:17:35.000000000 -0400
@@ -1,5 +1,4 @@
 #!./perl -w
-BEGIN { @INC = qw(lib) }
 use strict;
 use Config;
 use Getopt::Long;
diff -Naur perl-5.8.8.orig/installperl perl-5.8.8/installperl
--- perl-5.8.8.orig/installperl	2006-07-22 18:17:24.000000000 -0400
+++ perl-5.8.8/installperl	2006-07-22 18:17:35.000000000 -0400
@@ -3,8 +3,6 @@
 BEGIN {
     require 5.004;
     chdir '..' if !-d 'lib' and -d '../lib';
-    @INC = 'lib';
-    $ENV{PERL5LIB} = 'lib';
 }
 
 use strict;
diff -Naur perl-5.8.8.orig/lib/unicore/Makefile perl-5.8.8/lib/unicore/Makefile
--- perl-5.8.8.orig/lib/unicore/Makefile	2006-07-22 18:17:25.000000000 -0400
+++ perl-5.8.8/lib/unicore/Makefile	2006-07-22 18:17:35.000000000 -0400
@@ -1,11 +1,11 @@
 all:
-	../../miniperl -I../../lib ./mktables
+	../../miniperl-cross -I../../lib ./mktables
 
 TestProp.pl: mktables UnicodeData.txt Scripts.txt Blocks.txt PropList.txt
-	../../miniperl -I../../lib ./mktables -maketest
+	../../miniperl-cross -I../../lib ./mktables -maketest
 
 test:   TestProp.pl
-	../../miniperl -I../../lib TestProp.pl
+	../../miniperl-cross -I../../lib TestProp.pl
 
 clean:
 	rm -f *.pl */*.pl
diff -Naur perl-5.8.8.orig/Makefile.SH perl-5.8.8/Makefile.SH
--- perl-5.8.8.orig/Makefile.SH	2006-07-22 18:17:25.000000000 -0400
+++ perl-5.8.8/Makefile.SH	2006-07-22 18:17:35.000000000 -0400
@@ -591,7 +591,7 @@
 	$(CC) -o miniperl $(CLDFLAGS) \
 	    `echo $(obj) | sed 's/ op$(OBJ_EXT) / /'` \
 	    miniperlmain$(OBJ_EXT) opmini$(OBJ_EXT) perl$(OBJ_EXT) $(libs)
-	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
+	$(LDLIBPTH) ./miniperl-cross -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
 !NO!SUBS!
 		;;
 	next4*)
@@ -599,7 +599,7 @@
 miniperl: $& miniperlmain$(OBJ_EXT) $(LIBPERL) opmini$(OBJ_EXT)
 	$(CC) -o miniperl `echo $(obj) | sed 's/ op$(OBJ_EXT) / /'` \
 	    miniperlmain$(OBJ_EXT) opmini$(OBJ_EXT) perl$(OBJ_EXT) $(libs)
-	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
+	$(LDLIBPTH) ./miniperl-cross -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
 !NO!SUBS!
 		;;
 	darwin*)
@@ -620,7 +620,7 @@
 	-@rm -f miniperl.xok
 	$(CC) $(CLDFLAGS) $(NAMESPACEFLAGS) -o miniperl \
 	    miniperlmain$(OBJ_EXT) opmini$(OBJ_EXT) $(LLIBPERL) $(libs)
-	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
+	$(LDLIBPTH) ./miniperl-cross -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
 !NO!SUBS!
 		;;
 	*)
@@ -629,7 +629,7 @@
 	-@rm -f miniperl.xok
 	$(LDLIBPTH) $(CC) $(CLDFLAGS) -o miniperl \
 	    miniperlmain$(OBJ_EXT) opmini$(OBJ_EXT) $(LLIBPERL) $(libs)
-	$(LDLIBPTH) ./miniperl -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
+	$(LDLIBPTH) ./miniperl-cross -w -Ilib -MExporter -e '<?>' || $(MAKE) minitest
 !NO!SUBS!
 		;;
 	esac
@@ -769,38 +769,38 @@
 preplibrary: miniperl$(EXE_EXT) $(CONFIGPM) lib/lib.pm $(PREPLIBRARY_LIBPERL)
 	@sh ./makedir lib/auto
 	@echo "	AutoSplitting perl library"
-	$(LDLIBPTH) ./miniperl -Ilib -e 'use AutoSplit; \
+	$(LDLIBPTH) ./miniperl-cross -Ilib -e 'use AutoSplit; \
 		autosplit_lib_modules(@ARGV)' lib/*.pm
-	$(LDLIBPTH) ./miniperl -Ilib -e 'use AutoSplit; \
+	$(LDLIBPTH) ./miniperl-cross -Ilib -e 'use AutoSplit; \
 		autosplit_lib_modules(@ARGV)' lib/*/*.pm
 	$(MAKE) lib/re.pm
 
 lib/Config.pod: config.sh miniperl$(EXE_EXT) configpm Porting/Glossary
-	$(LDLIBPTH) ./miniperl -Ilib configpm --heavy=lib/Config_heavy.pl lib/Config.pm
+	$(LDLIBPTH) ./miniperl-cross -Ilib configpm --heavy=lib/Config_heavy.pl lib/Config.pm
 
 $(CONFIGPM): lib/Config.pod
 
 lib/ExtUtils/Miniperl.pm: miniperlmain.c miniperl$(EXE_EXT) minimod.pl $(CONFIGPM)
-	$(LDLIBPTH) ./miniperl minimod.pl > lib/ExtUtils/Miniperl.pm
+	$(LDLIBPTH) ./miniperl-cross minimod.pl > lib/ExtUtils/Miniperl.pm
 
 lib/re.pm: ext/re/re.pm
 	cp ext/re/re.pm lib/re.pm
 
 $(plextract):	miniperl$(EXE_EXT) $(CONFIGPM) x2p/s2p
 	@-rm -f $@
-	$(LDLIBPTH) ./miniperl -I`pwd`/lib $@.PL
+	$(LDLIBPTH) ./miniperl-cross -I`pwd`/lib $@.PL
 
 x2p/s2p: miniperl$(EXE_EXT) $(CONFIGPM) x2p/s2p.PL
 	cd x2p; $(LDLIBPTH) $(MAKE) s2p
 
 lib/lib.pm:	miniperl$(EXE_EXT) $(CONFIGPM)
 	@-rm -f $@
-	$(LDLIBPTH) ./miniperl -Ilib lib/lib_pm.PL
+	$(LDLIBPTH) ./miniperl-cross -Ilib lib/lib_pm.PL
 
 unidatafiles $(unidatafiles): uni.data
 
 uni.data: miniperl$(EXE_EXT) $(CONFIGPM) lib/unicore/mktables
-	cd lib/unicore && $(LDLIBPTH) ../../miniperl -I../../lib mktables -w
+	cd lib/unicore && $(LDLIBPTH) ../../miniperl-cross -I../../lib mktables -w
 	touch uni.data
 
 extra.pods: miniperl$(EXE_EXT)
@@ -829,7 +829,7 @@
 	no-install install.perl install.man install.html
 
 META.yml:	Porting/makemeta Porting/Maintainers.pl Porting/Maintainers.pm
-	$(LDLIBPTH) ./miniperl -Ilib Porting/makemeta
+	$(LDLIBPTH) ./miniperl-cross -Ilib Porting/makemeta
 
 install-strip:
 	$(MAKE) STRIPFLAGS=-s install DESTDIR="$(DESTDIR)"
@@ -857,17 +857,17 @@
 		cd ../pod; $(MAKE) compile; \
 	else :; \
 	fi
-	$(LDLIBPTH) ./perl installperl --destdir=$(DESTDIR) $(INSTALLFLAGS) $(STRIPFLAGS)
+	$(LDLIBPTH) ./miniperl-cross -Ilib installperl --destdir=$(DESTDIR) $(INSTALLFLAGS) $(STRIPFLAGS)
 	$(MAKE) extras.install
 
 install.man:	all installman
-	$(LDLIBPTH) ./perl installman --destdir=$(DESTDIR) $(INSTALLFLAGS)
+	$(LDLIBPTH) ./miniperl-cross -Ilib installman --destdir=$(DESTDIR) $(INSTALLFLAGS)
 
 # XXX Experimental. Hardwired values, but useful for testing.
 # Eventually Configure could ask for some of these values.
 install.html: all installhtml
 	-@test -f README.vms && cd vms && $(LNS) ../README.vms README_vms.pod && cd ..
-	$(LDLIBPTH) ./perl installhtml   \
+	$(LDLIBPTH) ./miniperl-cross -Ilib installhtml   \
       --podroot=. --podpath=. --recurse  \
       --htmldir=$(privlib)/html   \
       --htmlroot=$(privlib)/html  \
@@ -1173,7 +1173,7 @@
 # Targets for UTF16 testing:
 
 minitest.utf16: minitest.prep
-	- cd t && (rm -f perl$(EXE_EXT); $(LNS) ../miniperl$(EXE_EXT) perl$(EXE_EXT)) \
+	- cd t && (rm -f perl$(EXE_EXT); $(LNS) ../miniperl-cross$(EXE_EXT) perl$(EXE_EXT)) \
 		&& $(LDLIBPTH) ./perl TEST -minitest -utf16 base/*.t comp/*.t cmd/*.t run/*.t io/*.t op/*.t uni/*.t </dev/tty
 
 test.utf16 check.utf16: test_prep
@@ -1239,7 +1239,7 @@
 # Can't depend on lib/Config.pm because that might be where miniperl
 # is crashing.
 minitest: miniperl$(EXE_EXT) lib/re.pm minitest.prep
-	- cd t && (rm -f perl$(EXE_EXT); $(LNS) ../miniperl$(EXE_EXT) perl$(EXE_EXT)) \
+	- cd t && (rm -f perl$(EXE_EXT); $(LNS) ../miniperl-cross$(EXE_EXT) perl$(EXE_EXT)) \
 		&& $(LDLIBPTH) ./perl TEST -minitest base/*.t comp/*.t cmd/*.t run/*.t io/*.t op/*.t uni/*.t </dev/tty
 
 # Test via harness
diff -Naur perl-5.8.8.orig/MANIFEST perl-5.8.8/MANIFEST
--- perl-5.8.8.orig/MANIFEST	2006-07-22 18:17:25.000000000 -0400
+++ perl-5.8.8/MANIFEST	2006-07-22 18:17:35.000000000 -0400
@@ -33,15 +33,8 @@
 configure.gnu		Crude emulation of GNU configure
 cop.h			Control operator header
 Copying			The GNU General Public License
-Cross/config		Cross-compilation
 Cross/config.sh-arm-linux	Cross-compilation
-Cross/generate_config_sh	Cross-compilation
-Cross/installperl.patch		Cross-compilation
 Cross/Makefile		Cross-compilation
-Cross/Makefile.SH.patch	Cross-compilation
-Cross/README		Cross-compilation
-Cross/TODO		Cross-compilation
-Cross/warp		Cross-compilation
 cv.h			Code value header
 cygwin/cygwin.c		Additional code for Cygwin port
 cygwin/ld2.in		ld wrapper template for Cygwin port
diff -Naur perl-5.8.8.orig/pod/Makefile.SH perl-5.8.8/pod/Makefile.SH
--- perl-5.8.8.orig/pod/Makefile.SH	2006-07-22 18:17:25.000000000 -0400
+++ perl-5.8.8/pod/Makefile.SH	2006-07-22 18:17:35.000000000 -0400
@@ -63,7 +63,7 @@
 	    --podroot=.. --podpath=pod:lib:ext:vms \
 	    --libpods=perlfunc:perlguts:perlvar:perlrun:perlop
 
-PERL = ../miniperl
+PERL = ../miniperl-cross
 PERLILIB = $(PERL) -I../lib
 REALPERL = ../perl
 
diff -Naur perl-5.8.8.orig/utils/Makefile perl-5.8.8/utils/Makefile
--- perl-5.8.8.orig/utils/Makefile	2006-07-22 18:17:26.000000000 -0400
+++ perl-5.8.8/utils/Makefile	2006-07-22 18:17:35.000000000 -0400
@@ -1,5 +1,5 @@
 
-PERL = ../miniperl
+PERL = ../miniperl-cross
 REALPERL = ../perl
 
 # Files to be built with variable substitution after miniperl is
diff -Naur perl-5.8.8.orig/x2p/Makefile.SH perl-5.8.8/x2p/Makefile.SH
--- perl-5.8.8.orig/x2p/Makefile.SH	2006-07-22 18:17:26.000000000 -0400
+++ perl-5.8.8/x2p/Makefile.SH	2006-07-22 18:17:35.000000000 -0400
@@ -172,7 +172,7 @@
 # These should be automatically generated
 
 $(plextract):
-	../miniperl -I../lib $@.PL
+	../miniperl-cross -I../lib $@.PL
 
 find2perl: find2perl.PL
 
